inc/class-shortcake-bakery.php

```diff
@@ -117,4 +120,56 @@ public static function get_shortcake_admin_dependencies() {
 		return $r;
 	}
 
+	public function action_admin_enqueue_scripts() {
+		wp_enqueue_script( 'shortcake-bakery-admin', plugin_dir_url( dirname( __FILE__ ) ) . 'assets/js/shortcake-bakery-admin.js', array( 'media-views', 'shortcode-ui' ) );
+		wp_enqueue_style( 'shortcake-bakery', plugin_dir_url( dirname( __FILE__ ) ) . 'assets/css/shortcake-bakery.css' );
+
+		$strings = array(
+			'text' => array(
+				'addEmbed' => __( 'Insert Embed Code', 'shortcake-bakery' ),
+				'insertButton' => __( 'Insert embed', 'shortcake-bakery' ),
+				'customEmbedLabel' => __( 'Paste any custom embed code here. If it matches a known post element, that post element will be used rather than the embed code.', 'shortcake-bakery' ),
+				'noReversalMatches' => __( 'The embed code provided doesn\'t match any known post elements. This means that it may not display as expected.', 'shortcake-bakery' ),
+			),
+			'nonces' => array(
+				'customEmbedReverse' => wp_create_nonce( 'embed-reverse' ),
+			),
+		);
+
+		wp_localize_script( 'shortcake-bakery-admin', 'ShortcakeBakery', $strings );
+	}
+
+
+	/**
+	 * Output the "Add embed" button with WP media buttons.
+	 *
+	 * @return void
+	 */
+	public function action_media_buttons( $editor_id ) {
+		static $instance = 0;
+		$id_attribute = ( 1 === ++$instance ) ? 'insert-embed-button' : 'insert-embed-button-' . $instance;
+		printf( '<button type="button" id="%s" class="button insert-embed add_media" data-editor="%s"><span class="dashicons dashicons-editor-code"></span> %s</button>',
+			esc_attr( $id_attribute ),
+			esc_attr( $editor_id ),
+			esc_html__( 'Add Embed', 'fusion' )
+		);
+	}
+
+	public function action_ajax_shortcake_bakery_embed_reverse() {
+		if ( empty( $_POST['_wpnonce'] ) || empty( $_POST['custom_embed_code'] ) ) {
+			exit;
+		}
+
+		check_ajax_referer( 'embed-reverse', '_wpnonce' );
+
+		$provided_embed_code = wp_unslash( $_POST['custom_embed_code'] );
+		$reversal = $this->filter_pre_kses( $provided_embed_code );
```

Comment:
OK. For this UI, I'm considering a "success" if any change happens to the content. So I want to make sure to avoid normal kses behavior, so that WordPress stripping scripts out or texturizing isn't considered a successful reversal...